src/test/java/jewellery/inventory/integration/ProductCrudIntegrationTest.java

```diff
@@ -0,0 +1,232 @@
+package jewellery.inventory.integration;
+
+import jewellery.inventory.dto.request.ProductRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceQuantityRequestDto;
+import jewellery.inventory.dto.request.ResourceInUserRequestDto;
+import jewellery.inventory.dto.request.UserRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceRequestDto;
+import jewellery.inventory.dto.response.ProductResponseDto;
+import jewellery.inventory.dto.response.ResourcesInUserResponseDto;
+import jewellery.inventory.helper.ResourceTestHelper;
+import jewellery.inventory.helper.UserTestHelper;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Gemstone;
+import jewellery.inventory.repository.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+class ProductCrudIntegrationTest {
+
+    @Value(value = "${local.server.port}")
+    private int port;
+
+    private static final String BASE_URL_PATH = "http://localhost:";
+
+    private String getBaseUrl() {
+        return BASE_URL_PATH + port;
+    }
+
+    private String buildUrl(String... paths) {
+        return getBaseUrl() + "/" + String.join("/", paths);
+    }
+
+    private String getBaseResourceAvailabilityUrl() {
+        return buildUrl("resources", "availability");
+    }
+
+    private String getBaseResourceUrl() {
+        return "http://localhost:" + port + "/resources";
+    }
+
+    private String getBaseUserUrl() {
+        return "http://localhost:" + port + "/users";
+    }
+
+    private String getBaseProductUrl() {
+        return "http://localhost:" + port + "/products";
+    }
```

Comment:
These could make use of getBaseUrl method.

Reply:
Done!